src/reitit/core.cljc

```diff
@@ -60,69 +60,88 @@
   (cond->> (->> (walk data opts) (map-meta merge-meta))
            coerce (into [] (keep #(coerce % opts)))))
 
+(defn name-lookup [[_ {:keys [name]}] opts]
+  (if name #{name}))
+
+(defn find-names [routes opts]
+  (into [] (keep #(-> % second :name) routes)))
```

Comment:
Could be marginally more efficient to use a transducer. `(into [] (keep #(-> % second :name)) routes)`